src/api/controller/user_controller.py

```diff
@@ -38,3 +39,28 @@ async def register_user(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail=f"Invalid registration data: {str(e)}",
         )
+
+@router.get(
+    "/user/{id}",
+    response_model=UserResponse,
+    status_code=status.HTTP_200_OK,
+    responses={404: {"description": "User not found"}},
+)
+async def get_user_by_id(
```

Comment:
Since id is in parameter variable, we can skip having id in the method name as convention.

get_user_by_id(id: int) -> get_user(id: int)

Reply:
Changed

Reply:
Also changed in         user = await user_service.get_user(id)